mlrun/projects/project.py

```diff
@@ -2884,6 +2885,69 @@ def _init_function_from_obj(func, project, name=None):
     return name or func.metadata.name, func
 
 
+def _mask_credentials_in_function_object(function):
+    env_to_mask = [
+        "V3IO_ACCESS_KEY",
+        "V3IO_USERNAME",
+        "V3IO_PASSWORD",
+        "V3IO_API",
+        "V3IO_FRAMESD",
+        "AWS_ACCESS_KEY_ID",
+    ]
+    for env in function.spec.env:
+        # if env contain a value and not value_from, this means it wasn't masked then we remove it
+        # if env contain a value_from, this means it was masked then no need to remove it
+        env_name = get_item_name(env, "name")
+        env_value_from = get_item_name(env, "value_from")
+        if env_name in env_to_mask and not env_value_from:
+            set_item_attribute(env, "value", None)
+
+    if (
+        function.metadata.credentials
+        and function.metadata.credentials.access_key
+        and not function.metadata.credentials.access_key.startswith(
+            mlrun.model.Credentials.secret_reference_prefix
+        )
+    ):
+        function.metadata.credentials.access_key = None
+
+    _mask_volumes_in_function_object(function)
+
+    return function
+
+
+def _mask_volumes_in_function_object(function):
+    should_remove_v3io_mounts = False
+    volumes = []
+    for volume in function.spec.volumes:
+        if get_item_name(volume, "name") == "v3io":
```

Comment:
The `name` is indeed by default `v3io`, but it's not mandatory - the user can call `mount_v3io` and provide a different name for the volume.
The 100% way to detect the v3io flex volumes is to look for `flexVolume` volumes with `v3io/fuse` driver type.